### PR TITLE
BuildInfo tests without context

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Inject logger into BuildInfo methods, swap tests to testWithoutContext.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/build_info_test.dart [hunk 1]
```diff
@@ -3,58 +3,59 @@
...
+    testWithoutContext('CFBundleShortVersionString for iOS', () async {
+      String buildName = validatedBuildNameForPlatform(TargetPlatform.ios, 'xyz', logger);
       expect(buildName, isNull);
-      buildName = validatedBuildNameForPlatform(TargetPlatform.ios, '0.0.1');
+      buildName = validatedBuildNameForPlatform(TargetPlatform.ios, '0.0.1', logger);
       expect(buildName, '0.0.1');
-      buildName = validatedBuildNameForPlatform(TargetPlatform.ios, '123.456.xyz');
+
+      buildName = validatedBuildNameForPlatform(TargetPlatform.ios, '123.456.xyz', logger);
+      expect(logger.traceText, contains('Invalid build-name'));
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/build_info_test.dart:44]: As a treat.

File: packages/flutter_tools/test/general.shard/build_info_test.dart [hunk 2]
```diff
@@ -3,58 +3,59 @@
...
 
 void main() {
-  setUpAll(() { });
-
+  BufferLogger logger;
+  setUp(() {
+    logger = BufferLogger.test();
+  });
   group('Validate build number', () {
-    setUp(() async { });
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/build_info_test.dart:15]: 🤔

File: packages/flutter_tools/test/general.shard/build_info_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_info_test.dart:44]: 🍪

File: packages/flutter_tools/test/general.shard/build_info_test.dart [hunk 3]
```diff
@@ -3,58 +3,59 @@
...
-import '../src/context.dart';
 
 void main() {
-  setUpAll(() { });
-
+  BufferLogger logger;
+  setUp(() {
+    logger = BufferLogger.test();
+  });
   group('Validate build number', () {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/build_info_test.dart:16]: nit:

```suggestion

  group('Validate build number', () {
```

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**